stacksCollectible/index.ts

```diff
@@ -115,55 +102,67 @@ async function fetchNftData(
   return { collectionData, nftArray };
 }
 
-function organizeNFTsIntoCollection(
+export function organizeNFTsIntoCollection(
   collectionRecord: Record<string, StacksCollectionData>,
-  nftArray: Array<NonFungibleToken>,
+  nftArray: NonFungibleToken[],
   nftCollectionDataArray: Array<NftCollectionData | undefined>,
 ) {
+  const organized: Record<string, StacksCollectionData> = {};
+
   for (let i = 0; i < nftArray.length; i++) {
     const nft = nftArray[i];
     const principal: string[] = nft.asset_identifier.split('::');
     const contractInfo: string[] = principal[0].split('.');
     const contractId = principal[0];
 
-    if (contractInfo[1] === 'bns') continue; //collection already assigned
+    if (contractInfo[1] === 'bns') {
+      // currently stacks only supports 1 bns name per address
+      collectionRecord.bns = {
+        collection_id: contractId,
+        collection_name: 'BNS Names',
+        total_nft: 1,
+        all_nfts: [nft],
+      };
+    }
 
     const collectionData = nftCollectionDataArray[i];
 
-    //group NFTs into collections
-    if (collectionRecord[contractId]) {
-      const data = collectionRecord[contractId];
-
+    // group NFTs into collections
+    if (organized[contractId]) {
+      const data = organized[contractId];
       data.all_nfts.push(nft);
-      if (data.total_nft < 4) {
-        data?.thumbnail_nfts.push(nft);
-      }
-      data.total_nft += 1;
     } else {
-      collectionRecord[contractId] = {
+      organized[contractId] = {
         collection_id: contractId,
         collection_name: collectionData?.collection?.name ?? contractId,
         total_nft: 1,
         all_nfts: [nft],
-        thumbnail_nfts: [nft],
         floor_price: collectionData?.collection?.floorItem?.price
           ? microstacksToStx(new BigNumber(collectionData?.collection?.floorItem?.price)).toNumber()
           : 0,
       };
     }
   }
+
+  // sort and unique all_nfts
+  Object.values(organized).forEach((collection) => {
+    const sorted = collection.all_nfts.sort((a, b) => (a.identifier.tokenId < b.identifier.tokenId ? -1 : 1));
+    const unique = new Map(sorted.map((nft) => [nft.identifier.tokenId, nft]));
+    collection.all_nfts = Array.from(unique.values());
+    collection.total_nft = collection.all_nfts.length;
+  });
```

Comment:
nit: this probably won't change performance but is not better to remove duplicates first and then do the sort?